go.mod

```diff
@@ -1,70 +1,71 @@
 module github.com/NVIDIA/k8s-kata-manager
 
-go 1.20
+go 1.21
+
+toolchain go1.22.2
```

Comment:
Question -- does this mean that we suggest to build the project with go 1.22.2, but the minimum go version that can be used to build the module is 1.21?

Reply:
My understanding is that when we work with the go module for development-related things: it will use the `go1.22.2`  toolchain, but when another go project imports this module, they need to be on go 1.21 at the very least

From the [go documentation](https://go.dev/blog/toolchain):

> "There is also a new toolchain line in go.mod that specifies the minimum Go toolchain to use when working in a particular module. In contrast to the go line, toolchain does not impose a requirement on other modules. For example, a go.mod might say:"


> ```
> module m
> go 1.21.0
> toolchain go1.21.4
> ```
> This says that other modules requiring m need to provide at least Go 1.21.0, but when we are working in m itself, we want an even newer toolchain, at least Go 1.21.4.

Reply:
So the answer to your question would be - "yes"